tools/lib/h5diff.c

```diff
@@ -639,6 +639,20 @@ h5diff(const char *fname1, const char *fname2, const char *objname1, const char
         }
     }
 
+    if (opts->page_cache > 0) {
+        if ((fapl1_id = h5tools_get_fapl(fapl1_id, NULL, NULL)) < 0) {
```

Comment:
This should probably be assigned to a temporary variable first so that the old `fapl1_id` can be closed before reassigning `fapl1_id` to the temporary variable.

Reply:
Shouldn't that be done inside the h5tools_get_fapl function? Or at least add an option BOOL close_old_fapl.

Reply:
Every place I looked where h5tools_get_fapl() is being used the old is getting closed.

Reply:
Is there a reason not to close the previous fapl as I didn't find any current instances that continue to use it.

Reply:
I suggest just closing it inside and the future can add an option if necessary.

Reply:
I'd say the main reason is that it's probably surprising behavior to find that an ID you passed in to the function got closed from underneath you. It adds extra work too since you need to make an extra copy of your ID if you do plan to use it afterward.